iFixit/react-commerce
# Replace useInstantSearch with useMenu to prevent memory leak

closes #1012 
closes #1004 

This PR replaces `useInstantSearch` with `useMenu` for single selection facets. 
As of version v6.38.0 `useInstantSearch` appears to be flaky and causing issues such as the one described by #1004.
Moreover `useMenu` hook is better suited to model single select facets.

This change required quite a bit of refactoring because we need to distinguish between menu facets and refinement list facets higher up in the render tree (since even the routing state is affected). I took this as an opportunity to reorganise the components a little bit and fix other minor issues (such as #1012) in the process.

## QA

1. Open [Vercel preview](https://react-commerce-git-refactor-product-list-facets-ifixit.vercel.app/Parts)
2. Verify that #1004 is fixed
3. Verify that #1012 is fixed
4. Verify that there aren't regression in the behaviour of the facets accordion and the facets drawer when compared to production
